### PR TITLE
[0.2.3]Hydra migration 3 v2

### PR DESCRIPTION
**Pull request by rpartsey**
## Motivation and Context

- Pass on documentation + tutorials
- Make all DictConfig imports to be from omegaconf (use `if TYPE_CHECKING` as in registry.py)
- Remove YACS 
- Properly handle old checkpoint config trained with YACS
- README.mds with examples on how to use Hydra to configure Habitat/Baselines

This PR is the replacement of https://github.com/facebookresearch/habitat-lab/pull/996.

## Types of changes

- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by 0mdc** [comment]
Happy 1000th PR! 🥂

File: examples/tutorials/nb_python/Habitat_Lab.py [hunk 1]
```diff
@@ -418,7 +418,7 @@ def act(self, observations):
     pass
 
 # %% [markdown]
-# Deploy habitat-sim trained models on real robots with the [habitat-pyrobot bridge](https://github.com/facebookresearch/habitat-lab/blob/71d409ab214a7814a9bd9b7e44fd25f57a0443ba/habitat/sims/pyrobot/pyrobot.py)
+# Deploy habitat-sim trained models on real robots with the [habitat-pyrobot bridge](https://github.com/facebookresearch/habitat-lab/blob/main/habitat-lab/habitat/sims/pyrobot/pyrobot.py)
```
Comment (vincentpierre) [inline examples/tutorials/nb_python/Habitat_Lab.py:421]: I think we should keep a specific commit here (Does not need to be the same commit number) but this would help us make sure changes to main will not invalidate what the notebooks are saying.

File: habitat-baselines/habitat_baselines/rl/ver/README.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ The system has 5 components. 3 of these are the key components for learning:
 
 How these pieces are connected together is best seen via this diagram:
 
-![ver System Diagram](/habitat_baselines/rl/ver/images/ver-system.svg)
+![ver System Diagram](images/ver-system.svg)
```
Comment (vincentpierre) [inline habitat-baselines/habitat_baselines/rl/ver/README.md:16]: svg does not work I believe. We might have to export to PNG

File: habitat-lab/habitat/config/__init__.py [hunk 3]
```diff
@@ -4,123 +4,8 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
-r"""Habitat Lab Configuration
```
Comment (vincentpierre) [inline habitat-lab/habitat/config/__init__.py:7]: Was this comment used anywhere? Maybe it is used to generate documentation. Do you know @mathfac ?

File: habitat-lab/habitat/config/read_write.py [hunk 4]
```diff
@@ -1,15 +1,18 @@
...
+    from omegaconf import DictConfig
 
 
 # TODO : Delete this when migration to hydra is complete
 @contextmanager
-def read_write(config: Union[Node, Config]) -> Generator[Node, None, None]:
+def read_write(
+    config: Union[Node, "DictConfig"]
+) -> Generator[Node, None, None]:
     if isinstance(config, Node):
```
Comment (vincentpierre) [inline habitat-lab/habitat/config/read_write.py:16]: This can now be removed. There is no need to check here. We should remove anything below line 26 as this was for compatibility with YACS.

File: habitat-lab/requirements.txt [hunk 5]
```diff
@@ -1,6 +1,5 @@
 gym>=0.22.0,<0.23.1
 numpy>=1.20.0
-yacs>=0.1.8
```
Comment (vincentpierre) [inline habitat-lab/requirements.txt:3]: 👍

**Review by vincentpierre — APPROVED** [review]
Some minor suggestions

File: habitat-baselines/habitat_baselines/rl/ver/README.md (hunk 2 above)
Comment (vincentpierre) [reply habitat-baselines/habitat_baselines/rl/ver/README.md:16]: Ignore that comment, fixed it in [287bf45](https://github.com/facebookresearch/habitat-lab/pull/1000/commits/287bf45b51ba6266bb3d3149300d0ac38fe5c540)

**Comment by vincentpierre** [comment]
Dependent on #1007

File: habitat-lab/habitat/config/__init__.py (hunk 3 above)
Comment (rpartsey) [reply habitat-lab/habitat/config/__init__.py:7]: I haven't found it in the documentation.

File: habitat-baselines/habitat_baselines/agents/ppo_agents.py [hunk 6]
```diff
@@ -7,38 +7,42 @@
...
-    c.MODEL_PATH = "data/checkpoints/gibson-rgb-best.pth"
-    c.RESOLUTION = 256
-    c.hidden_size = 512
-    c.RANDOM_SEED = 7
-    c.PTH_GPU_ID = 0
-    c.goal_sensor_uuid = "pointgoal_with_gps_compass"
-    return c
+@dataclass
+class PPOAgentConfig:
+    INPUT_TYPE: str = "rgb"
```
Comment (mathfac) [inline habitat-baselines/habitat_baselines/agents/ppo_agents.py:29]: Do we still want to keep it uppercase here?

File: docs/conf.py [hunk 7]
```diff
@@ -32,8 +32,8 @@
     "registry",
```
Comment (mathfac) [inline docs/conf.py:32]: My feeling that we can enable more modules and classes to be enabled for doc building in separate PR.

**Review by mathfac — APPROVED** [review]
Thank you, that is a number of necessary changes across the all codebase.

File: habitat-baselines/habitat_baselines/agents/ppo_agents.py (hunk 6 above)
Comment (vincentpierre) [reply habitat-baselines/habitat_baselines/agents/ppo_agents.py:29]: I am the one that left it on purpose upper case. This is because there are some fields in the dataclass that are the same as the habitat_baselines config and I wanted to make sure users would know it is different. (I got confused before)
